hermes/serializers.py

```diff
@@ -138,59 +138,247 @@ class Meta(BaseTargetSerializer.Meta):
-    if date_format:
-        if 'jd' in date_format.lower():
-            try:
-                float(date)
-            except ValueError:
-                raise serializers.ValidationError({'date': _(f"Date: {date} does not parse. JD formatted dates must be a float value.")})
-        else:
-            try:
-                datetime.strptime(date, date_format)
-            except ValueError:
-                raise serializers.ValidationError({'date': _(f"Date: {date} does not parse based on provided date format: {date_format}.")})
-    else:
+def validate_date(date):
+    try:
+        float(date)
+    except ValueError:
         try:
             parse(date)
         except ValueError:
-            raise serializers.ValidationError({'date': _(f"Date: {date} does not parse with dateutil.parser.parse. Please specify a date_format or change your date.")})
+            raise serializers.ValidationError({'date': _(f"Date: {date} does not parse with dateutil.parser.parse. Please specify the date in a standard format or as a JD.")})
 
 
-class HermesMessageSerializer(serializers.Serializer):
-    title = serializers.CharField(required=True)
-    topic = serializers.CharField(required=True)
-    message_text = serializers.CharField(required=False, default='', allow_blank=True)
-    submitter = serializers.CharField(required=True)
-    authors = serializers.CharField(required=False, default='', allow_blank=True)
+class ReferenceDataSerializer(serializers.Serializer):
+    source = serializers.CharField(required=False)
+    citation = serializers.CharField(required=False)
+    url = serializers.URLField(required=False)
+
+    def validate(self, data):
+        validated_data = super().validate(data)
+        if not (validated_data.get('source') and validated_data.get('citation')):
+            if validated_data.get('source'):
+                raise serializers.ValidationError({
+                    'citation': 'Must set citation with source'
+                })
+            if validated_data.get('citation'):
+                raise serializers.ValidationError({
+                    'source': 'Must set source with citation'
+                })
+            if not validated_data.get('url'):
+                raise serializers.ValidationError({
+                    'source': 'Must set source/citation or url',
+                    'citation': 'Must set source/citation or url',
+                    'url': 'Must set source/citation or url',
+                })
+        return validated_data
 
+class OrbitalElementsSerializer(serializers.Serializer):
+    epoch_of_elements = serializers.CharField(required=True)
+    orbinc = serializers.FloatField(required=True, min_value=0.0, max_value=180.0)
+    longascnode = serializers.FloatField(required=True, min_value=0.0, max_value=360.0)
+    argofperih = serializers.FloatField(required=True, min_value=0.0, max_value=360.0)
+    eccentricity = serializers.FloatField(required=True, min_value=0.0, max_value=1.0)
```

Comment:
eccentricity technically doesn't have a max. Over 1 just means the object is unbound.